### ISSUE TITLE
Terra Tests Fail with Numpy 1.19.0

### ISSUE THREAD

**Issue by mtreinish** [Qiskit/qiskit-terra#4606]
### Information

- **Qiskit Terra version**: Master
- **Python version**: Any
- **Operating system**: Any

### What is the current behavior?

Since the release of numpy 1.19.0 CI started failing on 2 tests:
`test_qasm_2q_unitary`
https://github.com/Qiskit/qiskit-terra/blob/master/test/python/circuit/test_unitary.py#L248-L271
and
`test_pulse_limits:
https://github.com/Qiskit/qiskit-terra/blob/master/test/python/pulse/test_pulse_lib.py#L60-L89

It's not clear if this is a bug in numpy or a usage bug in terra. If it's an iss

This is being pinned in CI with https://github.com/Qiskit/qiskit-terra/pull/4599

### Steps to reproduce the problem

Run the tests with numpy 1.19.0

### What is the expected behavior?

The tests pass

### Suggested solutions

Find the source of the bug, and fix or report it upstream

**Comment by mtreinish** [comment]
The pulse side of the failures should be fixed with: https://github.com/Qiskit/qiskit-terra/pull/4613

**Comment by mtreinish** [comment]
Ok I've figured out where this is going wrong, but don't have a fix yet. Basically running:

```python
import numpy as np

from qiskit.quantum_info.synthesis import two_qubit_cnot_decompose

two_qubit_cnot_decompose(np.eye(4))
```

returns `U3(0, -pi/4, -pi/4)` for qubit 0 on windows with numpy 1.19.0 which causes the qasm tests to fail. While on linux and mac with numpy 1.19.0 it's still returning the correct `U3(0, 0, 0)`. We'll have to figure out where it's going wrong in the decomposition function to fix this, so it is definitely something we should fix before 0.15.0.
